org.elysium.parent/org.elysium.ui/src/org/elysium/ui/compiler/CompilerOutputProcessor.java

```diff
@@ -18,24 +22,25 @@
 public class CompilerOutputProcessor implements OutputProcessor {
 
 	/**
-	 * The file being compiled.
+	 * The problem parser for the file being compiled.
 	 */
-	private final IFile file;
+	private final ProblemParser problemParser;
 
 	/**
 	 * The console to which the compiler's output is written.
 	 */
 	private final LilyPondConsole console;
 
 	public CompilerOutputProcessor(IFile file, LilyPondConsole console) {
-		this.file = file;
 		this.console = console;
+		this.problemParser=new ProblemParser(file);
 	}
 
 	@Override
 	public void processOutput(String line) {
 		console.print(line);
-		IMarker problemMarker = ProblemParser.parse(file, line);
+		IMarker problemMarker = problemParser.parse(line);
+		final PdfAnnotation wsExternalIssue =problemParser.getWorkspaceExternalIssue();
```

Comment:
Using PdfAnnotation for storing a location in a source file is misleading for the developer. Please create a separate simple DTO for this purpose.

Reply:
I should have guessed ;-)
The separate DTO makes sense anyway, if we want to add an annotation to the opened editor.